evap/evaluation/templates/base.html

```diff
@@ -85,58 +83,78 @@
-                        dropdownParent: parent
-                    });
-                });
+            // used in derived templates. Tag inputs are multi-inputs where users can add options
+            const tagTomSelectOptions = {
+                create: true,
+                createOnBlur: true,
             };
-            applySelect2($('select').not('.form-template select').not('[disabled]').not('.no-select2').not('#id_delegate_to'));
-            // handle delegation modal, which includes a dropdown select field
-            applySelect2($('select#id_delegate_to'), $('#delegateSelectionModal'));
 
-            // don't use the placeholder text on disabled select2 elements
-            applySelect2Disabled = function(elem) {
-                elem.select2();
+            class MinimumInputLengthTomSelect extends TomSelect {
+                constructor(...args) {
+                    super(...args);
+                    this.settings.shouldLoad = () => true;  // Otherwise, for empty inputs, no dropdown would be shown.
+                }
+                search(query) {
+                    if(query.length < this.settings.minimumInputLength) {
+                        return this.sifter.prepareSearch(query, {});
+                    }
+                    return super.search(query);
+                }
+            }
+
+            applyTomSelect = function(elements, additionalOptions = {}) {
+                elements.forEach((element) => {
+                    const minimumInputLength = element.options.length >= 50 ? 3 : 0;
+
+                    element.tomselect?.destroy();
+                    element.classList.remove("form-select");  // TomSelect applies their own matching classes / styles
+                    if(element.parentElement.classList.contains("is-invalid")) {  // hotfix for our wrongly placed is-invalid
+                        element.parentElement.classList.remove("is-invalid");
+                        element.classList.add("is-invalid");
+                    }
+                    const baseOptions = {
+                        createOnBlur: true,
+                        placeholder: "{% trans 'Please select...' %}",
+                        hidePlaceholder: true,
+                        minimumInputLength,
+                        render: {
+                            option_create: (data, escape) => `<div class="create">${ escape(data.input) }</div>`,
+                            no_results: (data, escape) => {
+                                if(data.input.length < minimumInputLength) {
+                                    return `<div class="no-results">{% trans "Please enter ${ minimumInputLength } characters or more..." %}</div>`;
+                                } else {
+                                    return '<div class="no-results">{% trans "No results found" %}</div>';
+                                }
+                            }
+                        },
+                        plugins: {},
```

Comment:
```suggestion
```

Reply:
Required to be an object so we can use
```
baseOptions.plugins.clear_button = {"title": "{% trans 'Remove all items' %}"};
```
below -- or is there a smarter way to achieve that?